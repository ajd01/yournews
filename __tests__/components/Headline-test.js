import React from 'react'
import Overview from '../../components/Overview'
import renderer from 'react-test-renderer'
import {
  Context,
  InitialState
} from '../../store'

jest.mock('../../components/Tools', () => {
  return {
    __esModule: true,
    default: () => {
      return <div />
    }
  }
})

const article = {
  author: 'Mockauthor',
  publishedAt: 'MockpublishedAt',
  title: 'Mocktitle',
  urlToImage: 'MockurlToImage',
  description: 'Mockdescription',
  content: 'Mockcontent',
  url: 'Mockurl'
}

const value = {
  dispatch: jest.fn(),
  store: {
    ...InitialState,
    data: [{ ...article }]
  }
}

describe('Overview componente', () => {
  it('renders initial screen', () => {
    const tree = renderer.create(
      <Context.Provider value={value}>
        <Overview article={article} id={0} />
      </Context.Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
