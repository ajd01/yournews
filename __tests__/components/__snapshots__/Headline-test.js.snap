// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview componente renders initial screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <div>
    <View>
      <Text
        onClick={[Function]}
        style={
          Object {
            "fontSize": 22,
            "fontWeight": "bold",
            "padding": 5,
            "paddingBottom": 10,
            "paddingTop": 30,
            "width": 750,
          }
        }
      >
        Mocktitle
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "padding": 5,
              "width": 375,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
                "width": 750,
              }
            }
          >
            Mockauthor
          </Text>
          <Text
            style={
              Object {
                "fontSize": 15,
                "fontWeight": "bold",
                "width": 750,
              }
            }
          >
            Invalid Date
          </Text>
        </View>
        <View
          style={
            Object {
              "padding": 5,
              "width": 375,
            }
          }
        >
          <div />
        </View>
      </View>
      <Image
        onClick={[Function]}
        source={
          Object {
            "uri": "MockurlToImage",
          }
        }
        style={
          Object {
            "height": 750,
            "marginTop": 15,
            "paddingBottom": 5,
            "width": 750,
          }
        }
      />
    </View>
  </div>
</View>
`;
